routingconn.go

```diff
@@ -63,10 +63,8 @@ func dialRoutingConn(ctx context.Context, address string, opts *routingConnOptio
 	var transportDialOpt grpc.DialOption
 	var perRpcDialOpt grpc.DialOption
 
-	if opts.ClientCertificate != nil { // use tls
-		transportDialOpt = grpc.WithTransportCredentials(credentials.NewClientTLSFromCert(opts.ClientCertificate, ""))
-	} else if opts.InsecureSkipVerify { // use tls, but skip verification
-		creds := credentials.NewTLS(&tls.Config{InsecureSkipVerify: true})
+	if opts.ClientCertificate != nil || opts.InsecureSkipVerify {
```

Comment:
It would be ideal if we could rename this field from `ClientCertificate` to `RootCAs` or `ServerCertificate` or something else, "ClientCertificate" is very confusing since it's not the client certificate.